spec/frontend.test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import $ from 'jquery';
+
+// Components
+import PhotoBanner from '../client/components/PhotoBanner';
+import PhotoGrid from '../client/components/PhotoGrid';
+
+describe('PhotoBanner Test Suite', () => {
+  test('initial state of PhotoBanner', () => {
+    const wrapper = mount(<PhotoBanner />);
+    expect(wrapper.state('roomId')).toBe('5' || '10');
+    expect(wrapper.state('photos')).toHaveLength(0);
+  });
+
+  test('rendering of PhotoGrid upon available photos', () => {
+    const wrapper = mount(<PhotoBanner />);
+    const photos = [
+      { url: 'https://japaneseanimeinfo.up.seesaa.net/image/401-3e556.jpg' },
+      { url: 'https://i.pinimg.com/474x/47/14/f2/4714f2e99aef38240636d17e976f1672.jpg' },
+    ];
+
+    expect(wrapper.state('photos')).toHaveLength(0);
+    expect(wrapper.find('PhotoGrid')).toHaveLength(0);
+    wrapper.setState({ photos });
+    expect(wrapper.state('photos')).toHaveLength(2);
+    expect(wrapper.find('PhotoGrid')).toHaveLength(1);
+  });
+
+  // In-progress spy
+  test('links trigger an ajax call to getPhotosById', () => {
+    const wrapper = shallow(<PhotoBanner />);
+    const spy = jest.spyOn($, 'ajax');
+    wrapper.instance().getPhotosById();
+    expect(spy).toBeCalledWith({
+      method: 'GET',
+      url: '/api/photos/5',
+    });
+    expect(spy).toHaveBeenCalled();
+    spy.mockRestore();
+  });
+});
+
+describe('PhotoGrid Test Suite', () => {
+  test('should not find text that is not on the page', () => {
+    const photos = [{ url: 'https://japaneseanimeinfo.up.seesaa.net/image/401-3e556.jpg' }];
+    const wrapper = shallow(<PhotoGrid photos={photos} />);
+    expect(wrapper.find('text that is not on the page')).toHaveLength(0);
```

Comment:
What is this actually looking for? If nothing is marked as 'text that is not on the page', it's going to have a length of zero every time.

Reply:
I could likely have taken this out. In writing these initial test cases, I wanted to ensure the next result was not a false positive. Testing my test case, if you will.